data/src/main/java/com/microsoft/azure/kusto/data/auth/CloudInfo.java

```diff
@@ -56,9 +56,9 @@ public CloudInfo(boolean loginMfaRequired, String loginEndpoint, String kustoCli
         this.firstPartyAuthorityUrl = firstPartyAuthorityUrl;
     }
 
-    public static void manuallyAddToCache(String clusterUrl, CloudInfo cloudInfo) {
+    public static void manuallyAddToCache(String clusterUrl, CloudInfo cloudInfo) throws URISyntaxException {
         synchronized (cache) {
-            cache.put(clusterUrl, cloudInfo);
+            cache.put(UriUtils.setPathForUri(clusterUrl, ""), cloudInfo);
```

Comment:
nit: I'd think (new Uri(clusterUri)).ToString() would've been cleaner.

Reply:
Is it the same functionality? This removes everything after the domain

Reply:
Sorry, I didn't mean to include the `ToString()`.

I meant to create a `URI` object, which provides all subcomponents for free instead of having to reinvent the wheel.

Reply:
We need to strip the path out of the urls and make sure they're in the same format.
The only way to change the path is with the URI builder which is what we do in the method.

Do you mean change cache so that it would map URIs rather than strings? we could do that, but we still need UriUtils.setPathForUri

Reply:
Not changing what's cached, but instead how we get just the protocol+hostname.
It's a "nit" anyway, but I just see this as harder to read than using the `URL` object as the easy and standard way to get its components:
`
URL urlToClean = (new URL(clusterUrl));
cache.put(urlToClean.getProtocol() + "://" + urlToClean.getHost()) // not sure if port is relevant
`
I like `URIBuilder` as well: `new URIBuilder(uri).setPath("").build().toString()` - which is what's effectively done here when `UriUtils.setPathForUri()` is called.

I guess I'm just saying that I think it's less clear to me that `UriUtils.setPathForUri(clusterUrl, "")` simplifies the URL to just the protocol+hostname, compared to the other 2 approaches above.

Reply:
Maybe another method -  UriUtils.stripUriPath(clusterUrl) that will call UriUtils.setPathForUri(clusterUrl, "") ?

Reply:
I'd prefer to directly use the code that use built-in (or Apache) Java methods from my previous comment (`urlToClean.getHost()` or `new URIBuilder(uri).setPath("")`), instead of calling a static method. I don't even think that static method should exist. I just think this adds unnecessary complexity.

But this has lived longer than an average nit comment's lifespan, so nothing further from me :-)